client/rest/src/main/java/org/elasticsearch/client/RestClient.java

```diff
@@ -163,14 +163,15 @@ public synchronized void setHosts(HttpHost... hosts) {
      * {@link Exception#getCause()}.
      *
      * @param request the request to perform
+     * @param options options on the request to perform
      * @return the response returned by Elasticsearch
      * @throws IOException in case of a problem or the connection was aborted
      * @throws ClientProtocolException in case of an http protocol error
      * @throws ResponseException in case Elasticsearch responded with a status code that indicated an error
      */
-    public Response performRequest(Request request) throws IOException {
+    public Response performRequest(Request request, RequestOptions options) throws IOException {
```

Comment:
I would try not to touch these signatures. The idea could be to make Options a subclass of Request, and also an instance member of Request, so that the low-level REST client's methods remain the same, only to way to set the options changes (we haven't released yet the new request flavoured methods anyways).

Reply:
Agreed I didn't suggest that I meant that Request can hold an instance of Options.